packages/launchpad/src/use-tracking/index.tsx

```diff
@@ -0,0 +1,67 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { useEffect, useMemo } from 'react';
+import type { Task } from '../types';
+import type { SiteDetailsOptions } from '@automattic/data-stores';
+
+interface LogParams {
+	tasks: Task[];
+	checklistSlug: string | null;
+	context: string;
+	siteIntent: SiteDetailsOptions[ 'site_intent' ];
+}
+
+export const useTracking = ( params: LogParams ) => {
+	const { tasks, checklistSlug, context, siteIntent } = params;
+	const completedSteps = useMemo( () => tasks.filter( ( task ) => task.completed ), [ tasks ] );
+	const taskNames = useMemo( () => tasks.map( ( task ) => task.id ).join( ',' ), [ tasks ] );
+	const numberOfSteps = tasks.length;
+	const numberOfCompletedSteps = completedSteps.length;
+	const isCheckListCompleted = completedSteps.length === tasks.length;
+
+	useEffect( () => {
+		if ( tasks.length === 0 || ! siteIntent ) {
```

Comment:
Can I assume we always will have a site? So I can check if the site info was loaded instead of the site intent.

Reply:
We had this discussion in a separate thread as well (p1707419714859329-slack-C0Q664T29), but the key piece of information is that we may not have loaded the site details yet.

Reply:
Also, sites may not have an intent set, so it may be better to wait for the site details to be passed in as the prop so we can detect whether the details or site intent is null. And for now, I think we can always expect a site context, as the data and APIs for the task list are _always_ site-centric.

Reply:
I fixed it directly on the use-tracking as you can see here 266290ecb659fc4decc755e510a829d381b1bee0 to avoid introducing more changes on this PR but I think we can create a new issue to refactor the launchpad parent component to render the placeholder until we have the site info available.